consensus/service/src/config/tokens.rs

```diff
@@ -0,0 +1,416 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Tokens configuration.
+
+use crate::consensus_service::ConsensusServiceError;
+use mc_common::HashSet;
+use mc_consensus_enclave::FeeMap;
+use mc_transaction_core::{tokens::Mob, Token, TokenId};
+use serde::{Deserialize, Serialize};
+use std::{fs, iter::FromIterator, path::Path};
+
+/// Single token configuration.
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct TokenConfig {
+    /// Token ID.
+    token_id: TokenId,
+
+    /// Minimum fee.
+    minimum_fee: Option<u64>,
+}
+
+impl TokenConfig {
+    /// Token ID.
+    pub fn token_id(&self) -> TokenId {
+        self.token_id
+    }
+
+    /// Minimum fee.
+    pub fn minimum_fee_or_default(&self) -> Option<u64> {
+        self.minimum_fee
+            .or_else(|| FeeMap::default().get_fee_for_token(&self.token_id()))
+    }
+}
+
+/// Tokens configuration.
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct TokensConfig {
+    /// Token configurations (one for each supported token).
+    tokens: Vec<TokenConfig>,
```

Comment:
There's no `BTreeMap` of `no_std`, but I think the issue with that is serializing to JSON might break? I need to check. JSON doesn't allow integer keys in a map, only strings, so I am not sure how serde will behave with a `BTreeMap<u64, TokenConfig>`. Separately having the `token_id` be a part of the `TokenConfig` allows building the `minimum_fee_or_default()` method which is useful to have and used all across this PR. I think not having it would be a net negative.

Reply:
I agree about keeping `token_id` in `TokenConfig`. I'm thinking a `HashMap<TokenId, TokenConfig>` would be more convenient for lookups.

I tried it and it seems [serde_json correctly handles int keys](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=822431aae4603cc5d5ef23d5271e9f68)

Reply:
I think that would make the configuration less human-friendly though since you'll need to duplicate the token id - it will need to be present in both the key and the actual object, and that also means it can mismatch.